x-pack/elastic-agent/pkg/agent/program/testdata/constraints_config-filebeat.yml

```diff
@@ -4,7 +4,6 @@ filebeat:
     paths:
       - /var/log/hello1.log
       - /var/log/hello2.log
-    dataset: generic
```

Comment:
Shouldn't this become `dataset.name` instead of removing it?

Reply:
yes this became dataset.name thing is we injected it previously which i think is incorrect as we're providing processor to enrich event. if you think this information is helpful in a final beat config i will add it back but for now i filtered it out

Reply:
Even if the Beat does not understand it today, I think in the future the Beat should so having it there already would be helpful. Can happen in a later PR.